feat: Starknet support

piotr-roslaniec opened the pull request.

lrsaturnino commented on handlers/StarknetChainHandler.ts:
```diff
@@ -62,10 +345,346 @@ export class StarknetChainHandler extends BaseChainHandler<StarknetChainConfig>
+      logger.debug(`${logPrefix} L1 Contract Reveal Arg:`, revealData);
+
+      // Create the funding transaction array to match the contract expected format
+      const fundingTxStruct = [
```
initializeDeposit call may fail at the contract level because it's expected to receive a structured object instead of an array.

lrsaturnino commented on handlers/StarknetChainHandler.ts:
```diff
@@ -62,10 +345,346 @@ export class StarknetChainHandler extends BaseChainHandler<StarknetChainConfig>
+      ] as any;
+
+      // Take the first 5 elements of the reveal array
+      const revealArray = revealData.slice(0, 5) as any;
```
This is missing the 6th element (vault field) required by the contract's DepositRevealInfo struct:

1. fundingOutputIndex (uint32)
2. blindingFactor (bytes8)
3. walletPubKeyHash (bytes20)
4. refundPubKeyHash (bytes20)
5. refundLocktime (bytes4)
6. vault (address) ← missing

lrsaturnino commented on handlers/StarknetChainHandler.ts:
```diff
@@ -62,10 +345,346 @@ export class StarknetChainHandler extends BaseChainHandler<StarknetChainConfig>
+    try {
+      // First try to use the dynamic fee estimation from StarkGate
+      try {
+        const dynamicFee = await this.l1DepositorContract.estimateMessageFee();
```
This should be calling quoteFinalizeDepositDynamic(), which already includes buffer logic. If you want to call the StarkGate's estimateMessageFee() directly, you need to change the contract instance.

piotr-roslaniec commented:
@lrsaturnino Thanks for the review

I addressed the issues you've mentioned. In addition:
- For now, I added typed interfaces generated from `typechain` to the repo. I will figure out how to properly share those with `tbtc-v2` later.
- I also changed how `Reveal` data is passed around, including in the API (for consistency). This will potentially break something during integration.
- There is a critical vulnerability in one of our dependencies: https://github.com/advisories/GHSA-qhvj-jhf7-c63x. For some reason, I was unable to remove it from the dependency tree. Perhaps it's still somewhere in the supply chain. For that reason, I suggest we don't merge this yet.

lrsaturnino reviewed (approved):
LGTM

lrsaturnino commented on yarn.lock:
```diff
@@ -7294,11 +7470,41 @@ stack-utils@^2.0.3:
+  resolved "https://registry.yarnpkg.com/@starknet-io/types-js/-/types-js-0.7.10.tgz#d21dc973d0cd04d7b6293ce461f2f06a5873c760"
+  integrity sha512-1VtCqX4AHWJlRRSYGSn+4X1mqolI1Tdq62IwzoU2vUuEE72S1OlEeGhpvd6XsdqXcfHmVzYfj8k1XtKBQqwo9w==
+
+starknet@^6.24.1:
```
@piotr-roslaniec can we downgrade back to v6.0?
